pages/api/mock/[...slug].ts

```diff
@@ -286,13 +303,71 @@ export async function getMockedProperties(
     3
   );
 
+  console.log("Response:\n", openAiResponse);
+
   if (openAiResponse.length === 0) return { message: "Call to openai failed" };
 
-  const readded = addGPTdataToProperties(primitiveOnly, openAiResponse);
-  const newChunks = propertiesToChunks(Object.assign({}, transformed, readded));
+  if (isArray)
+    return rebuildPropertiesArray(
+      hardCodedProperties,
+      propertiesForAi,
+      openAiResponse,
+      allProperties
+    );
+  return rebuildProperties(
+    hardCodedProperties,
+    propertiesForAi,
+    openAiResponse,
+    allProperties
+  );
+}
+
+function rebuildProperties(
+  hardCodedProperties: Properties,
+  propertiesForAi: Properties,
+  openAiResponse: string,
+  transformed: Properties
+) {
+  const reAdded = addGPTdataToProperties(propertiesForAi, openAiResponse);
+  const newChunks = propertiesToChunks(
+    Object.assign({}, transformed, { ...reAdded, ...hardCodedProperties })
+  );
   return jsonReconstruct(newChunks);
 }
 
+function rebuildPropertiesArray(
+  hardCodedProperties: Properties,
+  propertiesForAi: Properties,
+  openAiResponse: string,
+  transformed: Properties
+) {
+  const arrayResponse = [];
+  // Loop through each array element
+  for (let idx = 0; idx < 3; idx++) {
+    // Stick the result from the gpt call back on
+    const readded = addGPTdataToProperties(
+      propertiesForAi,
+      openAiResponse,
+      idx
+    );
+
+    // hardcoded properties are a whole array, so we need to index
+    let hardcodedProperiesIdx: Properties = {};
+    for (let [key, value] of Object.entries(hardCodedProperties)) {
+      hardcodedProperiesIdx[key] = {
+        ...value,
+        data: value.data[idx],
+      };
+    }
+
+    const newChunks = propertiesToChunks(
+      Object.assign({}, transformed, { ...readded, ...hardcodedProperiesIdx })
+    );
+    arrayResponse.push(jsonReconstruct(newChunks));
+  }
+  return arrayResponse;
+}
```

Comment:
I'll be honest. In this file, there are so many data formats and functions manipulating between these formats that this function loses me.

Tests would make me feel 2x better about this :pray: